### PR TITLE
libutp: init at unstable-2018-05-15

### PR DESCRIPTION
**Pull request by onny**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libutp/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+, fetchFromGitHub
+, cmake}:
+
+stdenv.mkDerivation rec {
+  pname = "libutp";
+  version = "0.1";
+
+  src = fetchFromGitHub {
+    owner = "transmission";
+    repo = "${pname}";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libutp/default.nix:12]:
```suggestion
    repo = pname;
```

File: pkgs/development/libraries/libutp/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+, stdenv
+, fetchFromGitHub
+, cmake}:
+
+stdenv.mkDerivation rec {
+  pname = "libutp";
+  version = "0.1";
+
+  src = fetchFromGitHub {
+    owner = "transmission";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libutp/default.nix:11]:
```suggestion
    owner = "bittorrent";
```

There are no changes in transmission so far.

File: pkgs/development/libraries/libutp/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    repo = "${pname}";
+    rev = "fda9f4b3db97ccb243fcbed2ce280eb4135d705b";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libutp/default.nix:13]: https://github.com/bittorrent/libutp/issues/129

Also I think we want to fetch the latest commit here 2b364cbb0650bdab64a5de2abb4518f9f228ec44.

File: pkgs/development/libraries/libutp/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  meta = with lib; {
+    description = "uTorrent Transport Protocol library";
+    homepage = "https://github.com/bittorrent/libutp";
+    license = licenses.mit;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.onny ];
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libutp/default.nix:27]: One final new line to much.

File: pkgs/development/libraries/libutp/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake}:
+
+stdenv.mkDerivation rec {
+  pname = "libutp";
+  version = "0.1";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libutp/default.nix:8]: And this should be unstable-XXXX-XX-XX.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by onny** [comment]
Okay maybe it's ready now :)

File: pkgs/development/libraries/libutp/default.nix [hunk 6]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake}:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libutp/default.nix:4]:
```suggestion
, cmake
}:
```

File: pkgs/development/libraries/libutp/default.nix [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "libutp";
+  version = "unstable-2018-05-15";
+
+  src = fetchFromGitHub {
+    owner = "bittorrent";
+    repo = pname;
+    rev = "2b364cbb0650bdab64a5de2abb4518f9f228ec44";
+    sha256 = "0z94v27c4aki4giq7551g1gnycxvi2inm2rfjmzqkc34w0n9kyqa";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libutp/default.nix:15]:
```suggestion
    sha256 = "0yaiqksimnhwh14kmsq4kcyq6662b4ask36ni6p5n14dbyq1h2s6";
```

**Comment by SuperSandro2000** [comment]
@onny ofborg fails with 
``CMake Error: The source directory "/build/source" does not appear to contain CMakeLists.txt.``

**Comment by onny** [comment]
Okay should be good now :+1:

**Comment by onny** [comment]
By accident I lost access to the repository of this pull request. I created a new one [here](https://github.com/NixOS/nixpkgs/pull/132315).

**Pull request closed without merging**